src/main/host/descriptor/channel.c

```diff
@@ -131,6 +131,12 @@ static gssize channel_sendUserData(Transport* transport, Thread* thread, PluginV
         descriptor_adjustStatus((LegacyDescriptor*)channel, STATUS_DESCRIPTOR_WRITABLE, FALSE);
     }
 
+    /* The read-end of the channel is now readable if data is present */
+    if (channel->bufferLength > 0) {
+        descriptor_adjustStatus(
+            (LegacyDescriptor*)channel->linkedChannel, STATUS_DESCRIPTOR_READABLE, TRUE);
+    }
```

Comment:
I think this should already be done by the `channel_linkedWrite()` call above:

https://github.com/shadow/shadow/blob/1678d00cba5ef30c780efb67616a52a2b7e31828/src/main/host/descriptor/channel.c#L98-L99

Reply:
Oops, just saw this. Yeah this is one of the things I fixed. Rebasing now...